controllers/provisioners/eks/helpers.go

```diff
@@ -350,34 +380,49 @@ func (ctx *EksInstanceGroupContext) GetLabelList() []string {
 		isOverride = true
 		overrideLabels := strings.Split(val, ",")
 		for _, label := range overrideLabels {
-			labelList = append(labelList, label)
+			keyVal := strings.Split(label, "=")
+			labelMap[keyVal[0]] = keyVal[1]
 		}
 	}
 
 	if !isOverride {
 		// add default labels
-		labelList = append(labelList, fmt.Sprintf(RoleNewLabelFmt, instanceGroup.GetName()))
+		labelMap[RoleNewLabel] = instanceGroup.GetName()
 
 		// add the old style role label if the cluster's k8s version is < 1.16
 		clusterVersion := ctx.DiscoveredState.GetClusterVersion()
 		ver, err := semver.NewVersion(clusterVersion)
 		if err != nil {
 			ctx.Log.Error(err, "Failed parsing the cluster's kubernetes version", "instancegroup", instanceGroup.GetName())
-			labelList = append(labelList, fmt.Sprintf(RoleOldLabelFmt, instanceGroup.GetName()))
+			labelMap[fmt.Sprintf(RoleOldLabel, instanceGroup.GetName())] = ""
```

Comment:
Looks like we are changing the Role old label here from:
`"node-role.kubernetes.io/role=\"\""` to `"node-role.kubernetes.io/role"`
Should we use RoleOldLabelFmt?

Reply:
Nevermind - I guess since this is a map we no longer need the value